### PR TITLE
wappalyzer: Add export button and handle multi-select

### PR DESCRIPTION
**Pull request by kingthorin**
- CHANGELOG.md > Added change info.
- PopupMenuEvidence*.java > Added handling for multi-select (disable menu if multiple rows selected.)
- Techpanel > Add export button. Added icon to "Name" column which is now actually `Application` not string and uses a new custom renderer.
- TechTableModel > Removed icon column, made the new first column "Name" be based on `Application`.
- Messages.properties > Removed now un-used icon key.
- wappalyzer.html > Updated to note the multi-select behavior of the context menu, and mention the export button.

Signed-off-by: kingthorin <kingthorin@users.noreply.github.com>

### CODE REVIEW

File: addOns/wappalyzer/src/main/java/org/zaproxy/zap/extension/wappalyzer/TechPanel.java [hunk 1]
```diff
@@ -205,13 +223,15 @@ protected JXTable getTechTable() {
             techTable.setRowSelectionAllowed(true);
             techTable.setAutoCreateRowSorter(true);
             techTable.setColumnControlVisible(true);
+            TableCellRenderer renderer = new DefaultTableRenderer(new MappedValue(
+                    StringValues.EMPTY, IconValues.ICON), JLabel.CENTER);
+            techTable.setDefaultRenderer(ImageIcon.class, renderer);
```
Comment (kingthorin) [inline addOns/wappalyzer/src/main/java/org/zaproxy/zap/extension/wappalyzer/TechPanel.java]: I also tried this as:
```java
            TableColumn iconColumn = techTable.getColumn(Constant.messages.getString("wappalyzer.table.header.icon"));
            iconColumn.setCellRenderer(renderer);
```

Neither seems to work. It still exported or copy/pasted as the icon path (when there was an icon).

Comment (thc202) [reply addOns/wappalyzer/src/main/java/org/zaproxy/zap/extension/wappalyzer/TechPanel.java]: I'd suggest removing the icon from the table model, that belongs to the UI. The first column should return the `Application` (which `toString()` returns the name) then we can show the name and the icon in the same column and export/copy just the name.
The renderer would be:
```Java
TableCellRenderer renderer =
        new DefaultTableRenderer(
                new MappedValue(
                        StringValues.TO_STRING,
                        item -> {
                            if (item == null) {
                                return null;
                            }
                            Icon icon = ((Application) item).getIcon();
                            return icon != null ? icon : TRANSPARENT_ICON;
                        }),
                JLabel.LEADING);
techTable.setDefaultRenderer(Application.class, renderer);
```
with `TRANSPARENT_ICON` as, e.g.:
```Java
private static final Icon TRANSPARENT_ICON =
    new Icon() {

        @Override
        public void paintIcon(Component c, Graphics g, int x, int y) {
            // Nothing to do.
        }

        @Override
        public int getIconWidth() {
            return 32;
        }

        @Override
        public int getIconHeight() {
            return 32;
        }
    };
```

**Comment by kingthorin** [comment]
Thanks!

**Comment by kingthorin** [comment]
It's closer, toString on Application doesn't seem to return name just some memory reference.
image

Even if toString was name wouldn't that look kinda redundant? I'll push the current iteration maybe I'm not quite doing what you foresaw.

**Comment by thc202** [comment]
With removing the icon from the model was referring to the whole column, not just change the column name/type (the fist column would then be the one that's currently returning the application name). The `toString()` needs to be added (if you don't want to rely on `toString()` we need to change the renderer to use `getName()`).

**Comment by kingthorin** [comment]
Sorted, thanks for your help.

**Comment by thc202** [comment]
Thanks!

**Pull request merged**